### ISSUE TITLE
HttpBasicAuth doesn't function under Apache 2 CGI/FCGID

### ISSUE THREAD

**Issue by pcfreak30** [yiisoft/yii2#3500]
I have found that under FCGID in apache2, PHP_AUTH_USER and PHP_AUTH_PW $_SERVER elements do not exist. I looked at symphony, and it seems they are checking other headers as well due to this here: https://github.com/symfony/symfony/commit/1086faf0fb550d8f39a38a547e9e5210fb61203d

I have also tried using the apache2 rewrite rule "RewriteRule .\* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization},L]", and it seems to cause a 404 for pretty urls like /site/login.

Thanks.

**Comment by hbschr** [comment]
to get this thing rolling i started to work on this issue in my fork. i don't think it's ready yet, f.e. it only supports 'Basic', not 'DIgest'. and i think i added a new dependency with that `SetEnvIf`...

any comments?

p.s.: i'm away from internet in the next week, if you have questions be patient. zen.

**Comment by RSalo** [comment]
I have same problems with it. Maybe it's be due to with my working through corporative proxy server? Waiting for stable update with unit tests.

**Comment by RSalo** [comment]
Probably i've found my error. I've added this code:
```php
['access' => [
            'class' => AccessControl::className(),
            'only'  => ['view'],
            'rules' => [
                [
                    'actions' => ['view'],
                    'allow'   => true,
                    'roles'   => ['&'],
                ],
            ],
        ],
        ];
```
And works for me without errors!

**Comment by PommeVerte** [comment]
This actually affects all people using hhvm through apache.
It is possible to fix this by adding the following to the rewrite rule:
```
RewriteRule . index.php [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
```
And then bootstrapping the following to your entry script. (not sure if it's good practice but it works):

```php
if(isset($_SERVER['REDIRECT_HTTP_AUTHORIZATION']))
{
   list($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW']) =
  explode(':', base64_decode(substr($_SERVER['REDIRECT_HTTP_AUTHORIZATION'], 6)));
}
```

**Comment by rob006** [comment]
This was already fixed and documented, see  #13564.
